src/features/campaigns/hooks/useCampaignActivitiesFilters.ts

```diff
@@ -0,0 +1,61 @@
+import { ChangeEvent, useEffect, useMemo, useState } from 'react';
+
+import useLocalStorage from 'zui/hooks/useLocalStorage';
+import useDebounce from 'utils/hooks/useDebounce';
+import { ACTIVITIES } from '../types';
+
+const DEFAULT_FILTERS: ACTIVITIES[] = [
+  ACTIVITIES.CALL_ASSIGNMENT,
+  ACTIVITIES.AREA_ASSIGNMENT,
+  ACTIVITIES.SURVEY,
+  ACTIVITIES.TASK,
+  ACTIVITIES.EMAIL,
+];
+
+export default function useCampaignActivitiesFilters(
+  orgId: number,
+  campId?: number
+) {
+  const filtersKey = useMemo(
+    () => `campaignActivities:${orgId}:${campId ?? 'all'}:filters`,
+    [campId, orgId]
+  );
+  const searchKey = useMemo(
+    () => `campaignActivities:${orgId}:${campId ?? 'all'}:search`,
+    [campId, orgId]
+  );
+
+  const [filters, setFilters] = useLocalStorage<ACTIVITIES[]>(
+    filtersKey,
+    DEFAULT_FILTERS
+  );
+  const [storedSearchString, setStoredSearchString] = useLocalStorage<string>(
+    searchKey,
+    ''
+  );
+  const [searchString, setSearchString] = useState(storedSearchString);
+
+  useEffect(() => {
+    setSearchString(storedSearchString);
+  }, [storedSearchString]);
```

Comment:
If I am not mistaken, this is two states of the same thing:

If you look inside the `useLocalStorage` hook, you will see that there is already a state in there which is what is being returned and named by you here as `storedSearchString` and `setStoredSearchString`.

So the extra state here, `searchString` and `setSearchString` should not be neccessary - you can use the value and function returned from `useLocalStorage` directly!